SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -167,6 +167,74 @@ void TasmanianTridiagonalSolver::decompose(int n, double d[], double e[], double
     }
 }
 
+void TasmanianFourierTransform::dft(const int rank, const int n[], const std::complex<double> data[], std::complex<double> out[], const int direction) {
+    // Fourier transform code by ZBM
+    // DOES NOT NORMALIZE the outgoing Fourier coefficients
+
+    if(direction != 1 && direction != -1) std::cerr << "ERROR: direction must be -1 (forward) or 1 (backward); value is " << direction << std::endl;
+    int num_data = 1;
+    for(int r=0; r<rank; r++) {
+        num_data *= n[r];
+    }
+
+    std::complex<double> i(0,1);
```

Comment:
`i(0.0, 1.0)`

* With templates, you must be extra explicit when declaring thing, the compiler has to guess lots of stuff already and the chance for error is increased.